engine/src/engine.ts

```diff
@@ -191,14 +191,18 @@ export default class Engine {
     }
 
     const move = _move.trim();
+    let moveToShow = move;
+    if (move.split(" ")[1] == Command.Pass) {
+      moveToShow = move + " returning " + this.player(this.playerToMove).data.tiles.booster;
+    }
 
     if (!this.executeMove(move)) {
       assert(allowIncomplete, `Move ${move} (line ${this.moveHistory.length + 1}) is not complete!`);
       this.newTurn = false;
     }
 
     assert(this.turnMoves.length === 0, "Unnecessary commands at the end of the turn: " + this.turnMoves.join(". "));
-    this.moveHistory.push(move);
+    this.moveHistory.push(moveToShow);
```

Comment:
The move history is used in replay. Does it still work with `returning...` added?